### DISCUSSION TITLE
Does Open3D have any specific function to read .obj, .fbx or .blend files?

### DISCUSSION THREAD

**Discussion by whizbuzzer** [isl-org/Open3D#3836] [Q&A] [answered]
I initialized a TriangleMesh object using `open3d.geometry.TriangleMesh()`, and then read the .fbx file using `o3d.io.read_triangle_mesh()`, however, it returned the following in the console:

`[Open3D INFO] Skipping non-triangle primitive geometry of type: 12`

Is there any function that would allow me to read these files? I thought that there would be one in the library/package since the Open3D app does support FBX files: [http://www.open3d.org/2020/10/15/open3d-0-11-0/](url)

**Comment by germanros1987** [comment]
FBX should work. Could you please share your FBX model with us?

**Comment by errissa** [comment, marked as answer]
@whizbuzzer I confirmed that FBX files will load with Open3D. However, Open3D only supports triangle meshes and the warning indicates that your FBX file has polygons in it. Please try:

`o3d.io.read_triangle_mesh('my_model.fbx', enable_post_processing=True)`

which will run a post processing step during import that triangulates non-triangle polygons.
